### PR TITLE
Adds Search indexing and Job Queue

### PR DESCRIPTION
**Pull request by vinnyjth**
### What does this PR do, or why is it needed?

- Adds a background job queue and a "full index" worker that runs every day at 1:00. 

### What design trade-offs/decisions were made?

Punting delta indexing to another PR.

### How do I test this PR?

1. Clone this PR, `yarn install`
2. Uncomment this line: https://github.com/ApollosProject/apollos-prototype/pull/1062/files#diff-0f14ef6d2428a705a3936d9335d597f4R11
3. Be ready to open the Job Dashboard http://localhost:4000/admin/queues
4. Make sure you have redis running locally. 
5. Start the server
6. You should see one scheduled job and one completed job in the dashboard.

### What automated tests did you write?

Tests over datasource.

## TODO:

- [x] PR has a relevant title that will be understandable in a public changelog (ie...non developers)
- [ ] Closes [tag-issues-here]
- [x] No new warnings in tests, in storybook, and in-app
- [x] Upload GIF(s) of iOS and Android
- [x] Set a relevant reviewer

## REVIEW:

**Manual QA**

- [ ] Manual QA on iOS and ensure it looks/behaves as expected
- [ ] Manual QA on Android and ensure it looks/behaves as expected

**Code Review: Questions to consider**

- [ ] Read through the "Files changed" tab _very carefully_
- [ ] Edge cases: what assumptions are made about input?
- [ ] What kind of tests could be written?
- [ ] How might we make this easier for someone else to understand?
- [ ] Could the code be simpler?
- [ ] Will the code be easy to modify in the future?
- [ ] What's one part of these changes that makes you excited to merge it?

> The purpose of PR Review is to _improve the quality of the software._

### CODE REVIEW

File: packages/apollos-church-api/config.yml [hunk 1]
```diff
@@ -100,6 +100,7 @@ ROCK_MAPPINGS:
     - 8
   # Apollos will pull content items from these channels and expose them on the user feed
   # However, it will not expose the channel itself.
+  # These channels are also pulled when indexing searchable content.
```
Comment (conrad-vanl) [inline packages/apollos-church-api/config.yml]: Hmm....should we provide a different set of IDs for searching?

**Comment by conrad-vanl** [comment]
This PR is looking great! Everything worked for me locally. Lets make sure we get some real basic instructions for setting up and we can merge.

**Comment by vinnyjth** [comment]
👍 @conrad-vanl https://github.com/ApollosProject/apollos-prototype/blob/search-indexing/SEARCH.md

**Comment by vinnyjth** [comment]
Agh I hit merge before I saw your comment! I think that's a good idea, but I also like leaving it in the root because I think it's more likely to be seen by members of the team. Long term I'll migrate this to a link that points at a readme in the algolia folder :)
